### DISCUSSION TITLE
Relay and PWM

### DISCUSSION THREAD

**Discussion by gatoruss** [forkineye/ESPixelStick#579] [Q&A]
I have a D1 pro mini (Esp8266) with 16M.  I have espixelstick v 4,0 installed.  I can connect to the web, and on the home page see that I am connected to the d1 mini pro.  I do not see anywhere to configure pins for relays or pwm. I checked advance mode under the admin tab, but that doesn't seem to help.  What am I missing?

**Comment by MartinMueller2003** [comment]
When you enable relay support, a config sections is added to the devices screen. It lets you set the GPIO.

**Reply by kittenmcnuggets** [reply]
I got this far with my d1 mini, however all the gpio boxes are red with exclamation and I can't edit them... it is just the d1 with no other mb, looking g to just operate a 4 relay board @131.. thx for putting this all together btw

**Comment by MartinMueller2003** [comment]
I am not seeing such an issue:
![image](https://github.com/forkineye/ESPixelStick/assets/8608503/5bb0d59b-2a69-4f3f-960a-12cd1eaa6d5e)

**Comment by kittenmcnuggets** [comment]
![Screenshot_20231024-090407_Chrome](https://github.com/forkineye/ESPixelStick/assets/140961374/057ddd36-9eff-4308-8387-5e38ea07d977)
Ok, I also tried the latest posted ci, and am still getting it. Tried a couple of the d1 minis also. No other settings other than set output 2 to relay...

**Reply by kittenmcnuggets** [reply]
![Screenshot_20231024_094034](https://github.com/forkineye/ESPixelStick/assets/140961374/6c7f07f0-a10e-47d4-a330-88c87feafd5f)
Hardware is set to wemos d1 mini, it's a 8266 mini... tried with and without host name as I saw that was an issue a while ago (assuming it's not anymore)

Edit: wow, ok these cells were uneditable in chrome on my android phone. In chrome on windows, they all pre loaded with gpio "4294967292" in every field (but still red). From windows chrome I could edit and save....

**Comment by MartinMueller2003** [comment]
Yes they start as a -1 which is not valid. You need to set them to a value that is valid.

**Reply by kittenmcnuggets** [reply]
awesome, thanks for your patience. i have this pretty close to working but had two more q's if you are able... i have only ddp input and the 2nd relay output enabled. in xlights, if i apply the 4 relay models to the only open pixel port (1) in the visualizer, the test chase works fine cycling the relay states until either xlights auto uploads or i manual upload to espixelstick. then it forces espixelstick input 1 to be ws2811 and output 1 to be ws2811,and then the relays get bumped out and no longer respond... do i need to uncheck the upload features for the controller or is this something i'm setting incorrectly in the relay model? (i only see protocol options like ws2811 or dmx for the model, and they both cause this enabling output1 issue....

![image](https://github.com/forkineye/ESPixelStick/assets/140961374/8caaf05e-04d8-4299-a934-ad1194026426)

**Comment by MartinMueller2003** [comment]
Sorry but I am a vixen guy. You will need to go to the xLights forum to see how to deal with this.

**Reply by kittenmcnuggets** [reply]
no worries, thanks for the help. thought it might be something how the espixelstick handles the incoming uploads.

**Comment by MartinMueller2003** [comment]
Lots of changes lately on the xLights interface. Including downloading compressed files on ESP32 based platforms.
